src/csharp/Grpc.Core.Api/.gitignore

```diff
@@ -0,0 +1,2 @@
+bin
```

Comment:
Usually we would put .gitignore files in the root of the solution. Is there any other projects that require the outputs in bin/obj to be checked-in? The generated gRPC code maybe? Though I think in those scenarios, it's better to override the ignore behaviour with git add -f instead.